compiler-core/src/javascript/tests/functions.rs

```diff
@@ -362,3 +362,24 @@ pub fn version(n) {
 }"
     )
 }
+
+#[test]
+fn module_const_fn() {
+    assert_js!(
+        r#"
+pub fn int_identity(i: Int) -> Int { i }
+pub const int_identity_alias: fn(Int) -> Int = int_identity
```

Comment:
What happens if the const is higher up in the file than the function it references? Does it still generate valid to JavaScript?

Reply:
Good catch. It seems like when you assign a `Const::Var` to another module constant, it doesn't work.
```gleam
const x = y // this fails, saying that `y` is not in scope
const y = 5
```
However, when you assign a `Const::Var` to module function, it *does* work.
```gleam
const f_alias = f // this works
fn f(i: Int) -> Int { i }
```
This discrepancy feels like it could be a little confusing. I'm not sure how that part of the code works, but I assume that when a module is type checked it first sweeps through the function signatures so that you can use them out of order. Would it be difficult to add the module constants to this initial scope as well? I'd love to hear what you think about this.

I also think it could be error-prone to simply rely on the underlying behavior of javascript/erlang/whatever to be able to handle out of order usages/declarations of these global values. This is what this PR does currently -- it just declares the constants in the emitted code in the same order as they appear in the source.

Reply:
There is an issue for the ordering of constants limitation, so I don't think we need to worry about that here. We do need to make it so that this feature will always output valid JavaScript though, so we will need to do more than just inmates the statements in the order that we get them.

Reply:
Ok, so if the ordering of the constants isn't an issue, then I think we always output valid JS. The case you originally asked about, referring to a function in a constant definition before that function has been defined, does output valid JS as long as the JS codegen outputs functions using the `function` keyword due to hoisting. Is there an edge case that I'm missing?

Reply:
Brilliant, thank you. No edge case as far as I'm aware, it was a genuinely question as I was unsure.